### PR TITLE
Issue #30579537: Modernize

### PR DESCRIPTION
**Pull request by sposhe**
**Updates:**

- Splits directives and shared functions into discrete javascript modules
- Modernizes javascript
- Adds `teams` directive
- Changes `conf/conf.js` to `conf/conf.json` to avoid use of `eval()`

**Testing:** 

Note that this changes the configuration file from a javascript file to a JSON file, so your local configuration file will need to be changed to JSON.

Please test that each directive works as outlined in the README. (Note that to avoid pinging the team when testing `/savbot savasclaus`, the slack ID in `/js/savasclaus.js` should be set to your slack ID)

**Deployment:**

The configuration file on the server will also need to be changed to JSON.

### CODE REVIEW

**Comment by maddybvh** [comment]
Hi @sposhe , amazing work on this! 💪  👏   Everything works just as described.

I left a couple of code comments where I think the documentation could be more clear. In addition to those, I'd suggest we do the following for consistency with our other Node/JS work:
- Add `nodemon` for hot reloading
- Add style config files from our [JS Standards](https://github.com/savaslabs/js-standards) repo
- Save env variables as env variables, unless there's a compelling reason not to

If that all sounds good to you, I'm happy to push up those changes and pass this over to @dmurphy1 for deployment.

**Review by maddybvh — CHANGES_REQUESTED** [review]
Please see code comments!

File: conf/conf.example.json [hunk 1]
```diff
@@ -0,0 +1,34 @@
+{
+  "team": [
```
[the remark this answers is not in the document]
Comment (sposhe) [reply conf/conf.example.json:2]: It can be removed!

File: conf/conf.example.json [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+      "knownAs": "Dan",
+      "lastName": "Murphy",
+      "address": "A building\nBoston, MA 90210",
+      "slackID": "d.mulah",
+      "type": "remote"
+    },
+  ],
+  "port": 8080,
+  "command": "/savbot",
+  "token": "[String provided by Slack]",
```
[the remark this answers is not in the document]
Comment (sposhe) [reply conf/conf.example.json:30]: Just that that was how it was initially set up, and I don't have enough node experience to have used a `.env` file much!

File: README.md [hunk 3]
```diff
@@ -2,7 +2,7 @@ Slack integration to report the order of the weekly meeting.
 
 # Installation
 
-Copy `conf.example.js` into `conf.js` and add the Slack secret token from the Slack command configuration page (https://savaslabs.slack.com/services/B22JGF3GU). This is also where you would add new team members, or change the channel that the bot responds to.
+Duplicate `conf.example.json` as `conf.json` and add the Slack secret token from the Slack command configuration page (https://savaslabs.slack.com/services/B22JGF3GU). This is also where you would add new team members, or change the channel that the bot responds to.
```
[the remark this answers is not in the document]
Comment (sposhe) [reply README.md:5]: Agreed.

**Comment by sposhe** [comment]
All suggestions sound great, thanks!

**Review by maddybvh — APPROVED** [review]
Hi @dmurphy1 , I pushed up changes to address the change requests documented above, specifically:
- Adding `nodemon`
- Adding linting
- Converting `conf` to `.env` and accessing via `dotenv`.
- Updating documentation

Passing this over to you for a secondary review if you have time, or just to merge and deploy.

Note that during deployment, you'll need to set the env variables on production.

**Comment by dmurphy1** [comment]
LGTM, merging!

**Pull request merged**